src/lib/storage/index/abstract_table_index.hpp

```diff
@@ -18,8 +18,16 @@ namespace opossum {
  * (like and end-iterator), increments do not make a change, and equality comparisons with other instances of this class
  * always result in true.
  */
-class BaseTableIndexIterator : public std::iterator<std::forward_iterator_tag, const RowID> {
+class BaseTableIndexIterator {
  public:
+  using iterator_category = std::forward_iterator_tag;
+  using value_type = void;
+  using difference_type = std::ptrdiff_t;
+  using pointer = void;
+  using reference = RowID;
```

Comment:
This is not equivalent to the solution with the inheritance (see `Member types` in https://en.cppreference.com/w/cpp/iterator/iterator).

Reply:
value_type, pointer, and reference need to be changed.